File: src/no/omb/bigboat/Boat.java
package no.omb.bigboat;

public class Boat {

	private String sailNo;
	private String boatName;
	private String boatType;
	private String skipper;
	private String club;

	public Boat(String sailNo, String boatName, String boatType, String skipper, String club) {
		this.sailNo = sailNo;
		this.boatName = boatName;
		this.boatType = boatType;
		this.skipper = skipper;
		this.club = club;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result
				+ ((boatName == null) ? 0 : boatName.hashCode());
		result = prime * result + ((sailNo == null) ? 0 : sailNo.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Boat other = (Boat) obj;
		if (boatName == null) {
			if (other.boatName != null)
				return false;
		} else if (!boatName.equals(other.boatName))
			return false;
		if (sailNo == null) {
			if (other.sailNo != null)
				return false;
		} else if (!sailNo.equals(other.sailNo))
			return false;
		return true;
	}

	public String getSailNo() {
		return sailNo;
	}
	public void setSailNo(String sailNo) {
		this.sailNo = sailNo;
	}
	public String getBoatName() {
		return boatName;
	}
	public void setBoatName(String boatName) {
		this.boatName = boatName;
	}
	public String getBoatType() {
		return boatType;
	}
	public void setBoatTypee(String boatType) {
		this.boatType = boatType;
	}
	public String getSkipper() {
		return skipper;
	}
	public void setSkipper(String skipper) {
		this.skipper = skipper;
	}
	public String getClub() {
		return club;
	}
	public void setClub(String club) {
		this.club = club;
	}

	public String toString() {
		return getSailNo() + BigBoat.SEP + getBoatName() + BigBoat.SEP + getSkipper() + BigBoat.SEP + getBoatType() + BigBoat.SEP + getClub();
	}

}

File: src/no/omb/bigboat/data/Boat.kt
package no.omb.bigboat.data

import no.omb.bigboat.BigBoat
import java.util.*

data class Boat(
        val sailNo: String,
        val boatName: String,
        val boatType: String,
        val skipper: String,
        val club: String
) {
    override fun hashCode(): Int {
        val prime = 31
        var result = 1
        result = (prime * result
                + boatName.uppercase(Locale.getDefault()).hashCode())
        result = prime * result + sailNo.hashCode()
        return result
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null) return false
        if (javaClass != other.javaClass) return false
        val (sailNo1, boatName1) = other as Boat

        if (!boatName.equals(boatName1, ignoreCase = true)) return false
        return sailNo == sailNo1
    }

    override fun toString(): String {
        return sailNo + BigBoat.SEP + boatName + BigBoat.SEP + skipper + BigBoat.SEP + boatType + BigBoat.SEP + club
    }
}
